### PR TITLE
Add harvester for bicycle hiring in Barcelona

### PR DESCRIPTION
**Pull request by jmcanterafonseca**

### CODE REVIEW

**Comment by jmcanterafonseca** [comment]
@cclauss please could you review this PR?

File: Transportation/Bike/BikeHireDockingStation/harvest/bicycle_hire_station_barcelona_harvest.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
+#!bin/python
+# -*- coding: utf-8 -*-
+
+"""
+Harmonises data from the city of Barcelona corresponding to
+the bicycle hiring stations
+"""
+
+from __future__ import print_function
+import urllib2
```
Comment (cclauss) [inline Transportation/Bike/BikeHireDockingStation/harvest/bicycle_hire_station_barcelona_harvest.py]: See: https://stackoverflow.com/questions/2792650/python3-error-import-error-no-module-name-urllib2
```python
try:                 # Python 3
    from urllib.request import Request, urlopen
    from urllib.error import URLError
except ImportError:  # Python 2
    from urllib2 import Request, URLError, urlopen
```
Yes, it is ugly but it is correct for compatibility with both Python 2 and Python 3.  The alternative is to add a third party dependency such as [requests](http://docs.python-requests.org) or [six](https://pythonhosted.org/six).

File: Transportation/Bike/BikeHireDockingStation/harvest/bicycle_hire_station_barcelona_harvest.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+# Sanitize string to avoid forbidden characters by the Orion Broker
+
+
+def sanitize(str_in):
+    return re.sub(r"[<(>)\"\'=;-]", "", str_in)
+
+
+# Reads the data from the data source and returns a dictionary
+def read_data():
+    req = urllib2.Request(url=source)
```
Comment (cclauss) [inline Transportation/Bike/BikeHireDockingStation/harvest/bicycle_hire_station_barcelona_harvest.py]: Remove: `urllib2.` to align with the imports.  Repeat this removal on lines 50, 53, 57, 141, 144

Also you can simplify lines 51-61 down to:
```python
import contextlib  # move this to the top of the file

try:
    with contextlib.closing(urllib.urlopen(req)) as f:
        return f.read()  # return json data
except urllib2.URLError as e:
    print('Error while calling: %s : %s' % (source, e))
    return None
```

File: Transportation/Bike/BikeHireDockingStation/harvest/bicycle_hire_station_barcelona_harvest.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    station_list = parsed_data['stations']
+
+    ngsi_data = []
+
+    for station in station_list:
+        h_station = harmonize_station(station)
+        ngsi_data.append(h_station)
+
+    persist_data(ngsi_data)
+
```
Comment (cclauss) [inline Transportation/Bike/BikeHireDockingStation/harvest/bicycle_hire_station_barcelona_harvest.py]:
```python
def main():
    data = read_data()
    if data:
        persist_data(harmonize_station(station) for station in json.loads(data)['stations'])
    else:
        print("read_data() failed!')
```

File: Transportation/Bike/BikeHireDockingStation/harvest/bicycle_hire_station_barcelona_harvest.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        'Fiware-Servicepath': FIWARE_SERVICE_PATH
+    }
+
+    req = urllib2.Request(
+        url=(
+            DATA_BROKER +
+            '/v2/op/update'),
+        data=data_as_str,
+        headers=headers)
+    f = None
```
Comment (cclauss) [inline Transportation/Bike/BikeHireDockingStation/harvest/bicycle_hire_station_barcelona_harvest.py]: f = None  is not needed.

File: Transportation/Bike/BikeHireDockingStation/harvest/bicycle_hire_station_barcelona_harvest.py (hunk 3 above)
Comment (jmcanterafonseca) [reply Transportation/Bike/BikeHireDockingStation/harvest/bicycle_hire_station_barcelona_harvest.py]: too long line -1

Comment (jmcanterafonseca) [reply Transportation/Bike/BikeHireDockingStation/harvest/bicycle_hire_station_barcelona_harvest.py]: sorry but I still think code is clearer as it is now

**Comment by jmcanterafonseca** [comment]
thanks @cclauss for the review and useful comments

**Pull request merged**